File: accommodation_service/src/main/java/huy/project/accommodation_service/core/usecase/GetLanguageUseCase.java
package huy.project.accommodation_service.core.usecase;

import huy.project.accommodation_service.core.domain.dto.request.GetLanguageParams;
import huy.project.accommodation_service.core.domain.dto.response.PageInfo;
import huy.project.accommodation_service.core.domain.entity.LanguageEntity;
import huy.project.accommodation_service.core.port.ILanguagePort;
import lombok.RequiredArgsConstructor;
import org.springframework.data.util.Pair;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class GetLanguageUseCase {
    private final ILanguagePort languagePort;

    public Pair<PageInfo, List<LanguageEntity>> getAllLanguages(GetLanguageParams params) {
        return languagePort.getLanguages(params);
    }
}

File: location_service/LocationService.Core/UseCase/GetLanguageUseCase.cs
using LocationService.Core.Domain.Entity;
using LocationService.Core.Port;

namespace LocationService.Core.UseCase
{
    public interface IGetLanguageUseCase
    {
        Task<List<LanguageEntity>> GetLanguagesByIdsAsync(List<long> ids);
    }

    public class GetLanguageUseCase : IGetLanguageUseCase
    {
        private readonly ILanguagePort _languagePort;

        public GetLanguageUseCase(ILanguagePort languagePort)
        {
            _languagePort = languagePort;
        }

        public async Task<List<LanguageEntity>> GetLanguagesByIdsAsync(List<long> ids)
        {
            return await _languagePort.GetLanguagesByIdsAsync(ids);
        }
    }
}